src/examples/go_monorepo_workflows.yml

```diff
@@ -0,0 +1,180 @@
+            name: *build-test-api
+        - request_branch_deploy_dev:
+            type: approval
+            name: request-deploy
+            filters:
+              branches:
+                ignore:
+                  - master
+        - ric-orb/build_push_image:
+            <<: *cfg-isopod-version
+            context: dev
+            work_dir: *cfg-work-dir-api
+            name: docker
+            requires:
+              #when workflow is triggered for master branch request-deploy is not considered, because in that job there is filter that ignores master branch
+              - request-deploy
+              - *build-test-api
+        - ric-orb/deploy_job:
+            <<: *cfg-isopod-version
+            env: dev
+            context: dev
+            name: deploy-master-dev
+            work_dir: *cfg-work-dir-api
+            requires:
+              - docker
+        - ric-orb/deploy_job:
+            <<: *cfg-isopod-version
+            env: prod
+            context: prod
+            name: deploy-prod
+            work_dir: *cfg-work-dir-api
+            requires:
+              - deploy-master-dev
+
+    worker:
+      when:
+        or:
+          - << pipeline.parameters.worker >>
+          - << pipeline.parameters.common >>
+      jobs:
+        - ric-orb/go_build_test:
+            work_dir: *cfg-work-dir-worker
+            executor:
+              <<: *cfg-go-build-test-executor
+            additional_tests:
+              - run:
+                name: Running tests on common
+                command: make test
+                working_directory: common
+            name: *build-test-worker
+        - request_branch_deploy_dev:
+            type: approval
+            name: request-deploy
+            filters:
+              branches:
+                ignore:
+                  - master
+        - ric-orb/build_push_image:
+            <<: *cfg-isopod-version
+            context: dev
+            work_dir: *cfg-work-dir-worker
+            name: docker-branch
+            requires:
+              - request-deploy
+              - *build-test-worker
+        - ric-orb/deploy_job:
+            <<: *cfg-isopod-version
+            env: dev
+            context: dev
+            name: deploy-dev
+            work_dir: *cfg-work-dir-worker
+            requires:
+              - docker-branch
+        #master branch
+        - ric-orb/build_push_image:
+            <<: *cfg-isopod-version
+            context: prod
+            work_dir: *cfg-work-dir-worker
+            name: docker-master
+            filters:
+              branches:
+                only:
+                  - master
+            requires:
+              - *build-test-worker
+        - ric-orb/deploy_job:
+            <<: *cfg-isopod-version
+            env: dev
+            context: dev
+            name: deploy-master-dev
+            work_dir: *cfg-work-dir-worker
+            requires:
+              - docker-master
```

Comment:
what is the difference between a master- and a branch image?

i wonder mostly because you seem to deploy the master image to dev environment as well - so it works on all environments (or is this a mistake and you should deploy a dev-build for master branch to dev?)